### PR TITLE
Getting rid of varargs

### PR DESCRIPTION
**Pull request by Safrout1**
Varargs create problems when running nfs-ganesha on M1 Macs, resulting in null pointers. Thus, this pull request eliminates the existence of varargs.

### CODE REVIEW

**Comment by matvore** [comment]
You should squash the last two commits into one so this is easier to review and doesn't pollute the Git history with your generated cmake files.

Motivation for this PR: https://github.com/nfs-ganesha/ntirpc/issues/242
Precedent for this approach is in the macOS SDK (see Sept 17 comment in above issue)

**Comment by dang** [comment]
I don't see how this is possible.  Varargs are a standard part of C, and have been for decades.  There must be some way to support varargs on M1 macs.

**Review by dang — CHANGES_REQUESTED** [review]
Okay, I see the issue, it's not actually using varargs.  Can you squash all the commits together (or remove the extraneous ones), and make sure the commit message explains that it was casting a pointer to a non-varargs function to a pointer to a varargs function?

File: ntirpc/rpc/xdr.h [hunk 1]
```diff
@@ -257,7 +257,7 @@ typedef bool(*xdrproc_t) (XDR *, void *, u_int);
 /*
  * XXX can't actually prototype it, because some take three args!!!
  */
-typedef bool(*xdrproc_t) (XDR *, ...);
+typedef bool(*xdrproc_t) (XDR *, void *, unsigned int);
```
Comment (matvore) [inline ntirpc/rpc/xdr.h:260]: u_int will always be unsigned int, as far as I know. You could just remove the `#ifdef` conditional and always use this typedef (i.e. spell out `unsigned int`)

**Comment by Safrout1** [comment]
> Okay, I see the issue, it's not actually using varargs. Can you squash all the commits together (or remove the extraneous ones), and make sure the commit message explains that it was casting a pointer to a non-varargs function to a pointer to a varargs function?

Sorry for the inconvenience, I created another pull request: https://github.com/nfs-ganesha/ntirpc/pull/245

**Pull request closed without merging**